src/diffusers/schedulers/scheduling_heun_discrete.py

```diff
@@ -122,10 +144,16 @@ def index_for_timestep(self, timestep, schedule_timesteps=None):
 
         indices = (schedule_timesteps == timestep).nonzero()
 
-        if self.state_in_first_order:
-            pos = -1
+        # The sigma index that is taken for the **very** first `step`
+        # is always the second index (or the last index if there is only 1)
+        # This way we can ensure we don't accidentally skip a sigma in
+        # case we start in the middle of the denoising schedule (e.g. for image-to-image)
+        if len(self._index_counter) == 0:
```

Comment:
(Added by Patrick): I double checked that this works just like before by running img2img for SD 1.5 for different strength levels. 

Now that Shap-e adds a new exponential beta schedule that can lead to the same timestep being present more that 2 times we need to adapt this. Think an index counter dict works well here

Reply:
Thanks for checking!